modules/apps/product-navigation/product-navigation-user-personal-bar-web/src/main/resources/META-INF/resources/view.jsp

```diff
@@ -57,5 +57,103 @@
 		<span class="sign-in text-default" role="presentation">
 			<aui:icon cssClass="sign-in text-default" data="<%= anchorData %>" image="user" label="sign-in" markupView="lexicon" url="<%= themeDisplay.getURLSignIn() %>" />
 		</span>
+
+		<aui:script>
+			(function () {
+				var signInLink = document.querySelector('.sign-in > a');
+
+				if (!signInLink || signInLink.dataset.redirect !== 'false') {
+					return;
```

Comment:
Although, obviously, if you want an early `return`, you do. Just be aware that the client is going to get this code:

```js
(function () {
    (function () {
        // stuff in here
    })();
})();
```